### PR TITLE
Added default values for configuration, tests & some refactoring.

### PR DESCRIPTION
**Pull request by Livila**
Summary:
- [1922faf0334ea80427707b8266c75fdd962d5252] Improved regex speed in `OneFunction.java` by using a compiled pattern instead.
- [4852189be5def71d5f4806e13829fc1ef67ff280] Added a new value `RESULTS_FILE_LIMIT` for configuration.
- [4852189be5def71d5f4806e13829fc1ef67ff280] Added optional values when reading from the config for default values instead of crashes if a key is incorrect.
- [4852189be5def71d5f4806e13829fc1ef67ff280] Added tests for config.
- [4852189be5def71d5f4806e13829fc1ef67ff280] Added a table of default config values to `README.md`.
- [4852189be5def71d5f4806e13829fc1ef67ff280] `getString(...)` to get config value is renamed to `getProperty(...)`.
- [4852189be5def71d5f4806e13829fc1ef67ff280] Some refactoring and other minor changes.
- [de6d496c1f9de4b0bc67bbeee3366accc2d6d929] Added `LLM_THREADS` to configuration.
- [13bda16690bc2adc2e88c5860b9f2da180a30980] Changes, refactoring and a fix for `getInteger(...)`.
- [139a403f4154cefa50fe50ab87c302b6366b1fd5] Changed some logging from debug to trace. Added startup, shutdown and duration time.
- [2b940161ec8c76f74e8b6b8a1d76d12f189abcef] Fixed: #27
- [f0b836d6faba20d943eca7dc63eb5163e38b3821] Fixed: #23

### CODE REVIEW

File: src/main/java/CodeCheck/ConfigInterface.java [hunk 1]
```diff
@@ -54,36 +51,75 @@ public void loadConfig() throws IOException {
...
+        public String getProperty(String key) {
+            if (appProps == null) {
+                Log.warning("'%s' not initialized. The key '%s' was not loaded".formatted(appProps.getClass().getSimpleName(), key));
+                throw new RuntimeException("'appProps' is null - cannot continue. Configuration not initialized.");
+            }
+
+            try {
+                return appProps.get(key).toString().strip();
+            } catch (NullPointerException e) {
+                throw new NullPointerException(e.getMessage());
```
Comment (FeKozma) [inline src/main/java/CodeCheck/ConfigInterface.java:63]: Does this so anything?

File: src/main/java/CodeCheck/OneFunction.java [hunk 2]
```diff
@@ -2,41 +2,52 @@
...
     public Comparison compare(OneFunction other, LLM llm) {
-        switch (equals(other)) {
-            case 1: return new Comparison(true, "Functions are identical").setFunctions(this, other);
-            case 2: return similar(other, llm);
-            case 3: return new Comparison(false, false, "Empty or closetherby");
-            default:
-                return new Comparison();
-        }
+        return switch (equals(other)) {
+            case 1 -> new Comparison(true, "Functions are identical.").setFunctions(this, other);
```
Comment (FeKozma) [inline src/main/java/CodeCheck/OneFunction.java:33]: One should have enums here instead

**Review by FeKozma — APPROVED** [review]

File: src/main/java/CodeCheck/ConfigInterface.java (hunk 1 above)
Comment (Livila) [reply src/main/java/CodeCheck/ConfigInterface.java:63]: Related issue: #28.

File: src/main/java/CodeCheck/OneFunction.java (hunk 2 above)
Comment (Livila) [reply src/main/java/CodeCheck/OneFunction.java:33]: Related issue: #29.

**Review by Livila — COMMENTED** [review]
Issues created for further improvements.

**Pull request merged**